macros/src/types/newtype.rs

```diff
@@ -38,17 +39,26 @@ pub(crate) fn newtype(
         _ => {}
     };
 
-    let inner_ty = &inner.ty;
+    if let (Some(_type_as), Some(_type_override)) = (&type_as, &type_override) {
+        syn_err!("`type` is not compatible with `as`")
+    }
+
+    let inner_ty = if let Some(_type_as) = &type_as {
```

Comment:
Rename the variable to remove the `_` prefix, and if possible use `ref` instead of `&`